lib/winston/transports/http.js

```diff
@@ -177,13 +177,17 @@ module.exports = class Http extends TransportStream {
     delete options.path;
 
     // Prepare options for outgoing HTTP request
+    const headers = Object.assign({}, this.headers);
+    if (auth && auth.bearer) {
+      headers['Authorization'] = `Bearer ${auth.bearer}`;
+    }
     const req = (this.ssl ? https : http).request({
       method: 'POST',
       host: this.host,
       port: this.port,
       path: `/${path.replace(/^\//, '')}`,
-      headers: this.headers,
-      auth: auth ? (`${auth.username}:${auth.password}`) : '',
+      headers: headers,
+      auth: (auth && auth.username && auth.password) ? (`${auth.username}:${auth.password}`) : '',
```

Comment:
Now we're processing basic auth in one place but bearer auth in another.  Can we just make a helper function or something that processes any type of auth, in a single place?  (e.g. AccessToken would be another type...)

Reply:
I would have liked the same way to process bearer and basic also, but the way it's implemented right now would require some new dependency.

Based on [http-request](https://nodejs.org/api/http.html#http_http_request_url_options_callback) , when you add the "auth" property in http.request, that is automatically MD5ed and added as a header.
There is no similar alternative for bearer unfortunately.

Now, if you want to have the same way for basic and bearer, I can easily add the Authorization header for basic next to bearer, but i would require a reference to crypto.MD5 (or some MD5 package).

Let me know what your thoughts are.